Dockerfile.assisted-service.ocp

```diff
@@ -1,5 +1,5 @@
 # Build binaries
-FROM registry.ci.openshift.org/openshift/release:golang-1.17 as builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.18-openshift-4.11 AS builder
```

Comment:
@lranjbar is it ok that this says 4.11 and not 4.12? you mentioned this is intended for 4.12 but I don't really understand what's going on here